02-Generics/Generics/Generics.cs

```diff
@@ -1,174 +1,113 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Net;
+using System.Text;
 
 namespace Task.Generics {
 
 	public static class ListConverter {
 
-		private static char ListSeparator = ',';  // Separator used to separate values in string
-
-		/// <summary>
-		///   Converts a source list into a string representation
-		/// </summary>
-		/// <typeparam name="T">type  of list items</typeparam>
-		/// <param name="list">source list</param>
-		/// <returns>
-		///   Returns the string representation of a list 
-		/// </returns>
-		/// <example>
-		///   { 1,2,3,4,5 } => "1,2,3,4,5"
-		///   { '1','2','3','4','5'} => "1,2,3,4,5"
-		///   { true, false } => "True,False"
-		///   { ConsoleColor.Black, ConsoleColor.Blue, ConsoleColor.Cyan } => "Black,Blue,Cyan"
-		///   { new TimeSpan(1, 0, 0), new TimeSpan(0, 0, 30) } => "01:00:00,00:00:30",
-		/// </example>
+		private static char ListSeparator = ',';  
+
 		public static string ConvertToString<T>(this IEnumerable<T> list) {
-			// TODO : Implement ConvertToString<T>
-			throw new NotImplementedException();
+			if (list == null)
+				throw new ArgumentNullException("List is null");
+
+			StringBuilder stringBuilder = new StringBuilder();
+			foreach (var item in list)
+				stringBuilder.Append(item.ToString() + ListSeparator);
+
+			return stringBuilder.ToString().Remove(stringBuilder.Length - 1);
 		}
 
-		/// <summary>
-		///   Converts the string respresentation to the list of items
-		/// </summary>
-		/// <typeparam name="T">required type of output items</typeparam>
-		/// <param name="list">string representation of the list</param>
-		/// <returns>
-		///   Returns the list of items from specified string
-		/// </returns>
-		/// <example>
-		///  "1,2,3,4,5" for int => {1,2,3,4,5}
-		///  "1,2,3,4,5" for char => {'1','2','3','4','5'}
-		///  "1,2,3,4,5" for string => {"1","2","3","4","5"}
-		///  "true,false" for bool => { true, false }
-		///  "Black,Blue,Cyan" for ConsoleColor => { ConsoleColor.Black, ConsoleColor.Blue, ConsoleColor.Cyan }
-		///  "1:00:00,0:00:30" for TimeSpan =>  { new TimeSpan(1, 0, 0), new TimeSpan(0, 0, 30) },
-		///  </example>
 		public static IEnumerable<T> ConvertToList<T>(this string list) {
-			// TODO : Implement ConvertToList<T>
-			// HINT : Use TypeConverter.ConvertFromString method to parse string value
-			throw new NotImplementedException();
-		}
+			if (list == null)
+				throw new ArgumentNullException("List is null");
+
+			var result = new List<T>();
+			foreach (var item in list.Split(ListSeparator))
+				result.Add((T)TypeDescriptor.GetConverter(typeof(T)).ConvertFromString(item));
 
+			return result;
```

Comment:
u can use yield return
https://metanit.com/sharp/tutorial/4.12.php

Reply:
TypeDescriptor.GetConverter(typeof(T))
We can use one converter for all elements of the collection. Let's declare a local variable and assign the value of the converter to it. Further in the foreach we will use this local variable
(Now the converter is calculated for each element of the collection)